cinnabar/util.py

```diff
@@ -97,16 +97,21 @@ def init_logging():
 
 
 class ConfigSetFunc(object):
-    def __init__(self, key, values, extra_values=()):
+    def __init__(self, key, values, extra_values=(), default='', remote=None):
         self._config = None
         self._key = key
         self._values = values
         self._extra_values = extra_values
+        self._default = default
+        self._remote = remote
 
     def __call__(self, name):
         if self._config is None:
             from .git import Git
-            config = Git.config(self._key) or ''
+            if self._remote:
+                config = Git.config(self._key, self._remote) or self._default
```

Comment:
Ideally the error messages further below would be adapted to show what actual variable was used, which the API doesn't currently allow you to know.

Reply:
Yeah, I don't see a clean way to bubble that up right now